mirrord/agent/src/steal/connections.rs

```diff
@@ -304,9 +312,21 @@ impl StolenConnections {
                 rx: task_rx,
             };
 
+            if filtered {
+                STEAL_FILTERED_CONNECTION_SUBSCRIPTION.inc();
+            } else {
+                STEAL_UNFILTERED_CONNECTION_SUBSCRIPTION.inc();
+            }
+
             match task.run().await {
                 Ok(()) => {
                     tracing::trace!(connection_id, "Connection task finished");
+
+                    if filtered {
+                        STEAL_FILTERED_CONNECTION_SUBSCRIPTION.dec();
+                    } else {
+                        STEAL_UNFILTERED_CONNECTION_SUBSCRIPTION.dec();
+                    }
```

Comment:
I think you need to handle these counters elsewhere (maybe `ConnectionTask`?). Even if we have a filtered port subscription, the connection might still be rejected as not HTTP (and proxied raw to the original target). See `ConnectionTask::run` code

Reply:
Think it's in the right place now?